File: testdata/bench/dispatch.py
handlers = {
    "add": lambda x, y: x + y,
    "mul": lambda x, y: x * y,
    "sub": lambda x, y: x - y,
}

events = []
n = 200000

for i in range(n):
    op = "add" if i % 3 == 0 else ("mul" if i % 3 == 1 else "sub")
    events.append({
        "op": op,
        "a": i,
        "b": i % 7,
    })

out = 0
for e in events:
    fn = handlers[e["op"]]
    out += fn(e["a"], e["b"])

print(out)

File: testdata/bench/dispatch.lua
local handlers = {
    add = function(x, y) return x + y end,
    mul = function(x, y) return x * y end,
    sub = function(x, y) return x - y end,
}

local events = {}
local n = 200000

for i = 0, n - 1 do
    local op = "sub"
    if i % 3 == 0 then
        op = "add"
    elseif i % 3 == 1 then
        op = "mul"
    end
    table.insert(events, {
        op = op,
        a = i,
        b = i % 7,
    })
end

local out = 0
for _, e in ipairs(events) do
    local fn = handlers[e.op]
    out = out + fn(e.a, e.b)
end

print(out)
